frontend/src/entities/group/create/group-create.tsx

```diff
@@ -0,0 +1,217 @@
+/**
+ Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License").
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import React, { useMemo, useState } from 'react';
+import { z } from 'zod';
+import { issuesToErrors, useValidationReducer } from '../../../shared/validation';
+import { useAppDispatch } from '../../../config/store';
+import NotificationService from '../../../shared/layout/notification/notification.service';
+import { useNavigate, useParams } from 'react-router-dom';
+import { Button, Container, FormField, Header, Input, SpaceBetween, Textarea, } from '@cloudscape-design/components';
+import Form from '@cloudscape-design/components/form';
+import ContentLayout from '../../../shared/layout/content-layout';
+import { createGroup, getGroup, updateGroup } from '../group.reducer';
+
+export type GroupCreateProperties = {
+    isEdit?: boolean;
+};
+
+export function GroupCreate ({isEdit}: GroupCreateProperties) {
+    const dispatch = useAppDispatch();
+    const notificationService = NotificationService(dispatch);
+    const navigate = useNavigate();
+    const {groupName} = useParams();
+    const [initialLoaded, setInitialLoaded] = useState(false);
+
+    const groupSchema = z.object({
+        name: z
+            .string()
+            .min(3, {
+                message: 'Group name must be longer than 3 characters',
+            })
+            .max(24, {
+                message: 'Group name cannot be more than 24 characters',
+            })
+            .regex(/^[a-zA-Z0-9]+$/, {
+                message: 'Group name can only contain alphanumeric characters.',
+            }),
+        description: z
+            .string({
+                invalid_type_error: 'Group description must be a string',
+            })
+            .min(1, {
+                message: 'Group description cannot be empty',
+            })
+            .max(4000, {
+                message: 'Group description cannot be more than 4000 characters.',
+            })
+            .regex(/[ -~]/, {
+                message: 'Group description can only contain printable characters',
+            }),
```

Comment:
IMHO I think this convention is very confusing to read. It looks like ` `, `-`, or `~`. But barring that I think this should probably be `/^[ -~]$/` so every character is not a printable character.

Reply:
Sure, no preference from me as long as the message is clear. I updated. 

I had copied this from the project description in the `project-create.tsx` file.